### ISSUE TITLE
Find "new movies in theatre" or just new movies 

### ISSUE THREAD

**Issue by Seanmusse** [cinemagoer/cinemagoer#232]
#### Issue description

Hey there! I would like to find movies that have been released recently, is there a way to do this is imdbpy? Ideally I'd like to be able to find movies in [this](https://www.imdb.com/movies-in-theaters/?ref_=nv_mv_inth) link. Read through the docs but couldn't find anything, just top 250 which isn't what I'm looking for.

Edit: I'd like to thank the developer(s), great project :)

#### Version of IMDbPY and Python

- **Python:** `Python 3.6.8`
- **IMDbPY:** `6.9dev20190822`

**Comment by alberanid** [comment]
Hi,
there's no parser for that page.

It should not be too complex to write (and you may look at the simpler parsers in movieParser.py), but some things must be considered - first of all: new movies in theaters... ok, but WHERE? Is it possible to select the country? It's just done using your IP address or the language set by the browser?

**Comment by kostyafarber** [comment]
Hi I'd look to contribute to this. Any recommendations on where to get started?

**Comment by alberanid** [comment]
@kostyafarber thanks!

It's not so simple, since it's not just another page about a movie, but let's see...

In the `IMDbBase` class (`imdb/__init__.py` file) add a new key named `showtimes` that points to https://www.imdb.com/showtimes in the self.urls dict (see the `set_imdb_urls` method).

In the same class, add a `get_showtimes` method with the needed arguments (genre? date? other stuff that can be filtered / searched for) and just raise a `NotImplementedError` here.

Now you have to edit the `IMDbHTTPAccessSystem` class in `imdb/parser/http/__init__.py`.

In this class, override the `get_showtimes` method: it should take the passed arguments, and use them with the self.urls['showtimes'] path to build the complete URL for the query.

At the top of this file, you need to import a new module, let's call it `showtimesParser`; you also need to add a new "_ModuleProxy" for it (see the `__init__` method and do the same thing).

Back to the `get_showtimes` method: do the query calling `self._retrieve(url)`. Pass the result to `self.showtimesParser.showtime_parser`. See similar methods in this class.

Create the `imdb/parser/http/showtimesParser.py`; you can copy the topBottomParser.py file and use it as an example.

Basically:
* create DOMHTMLShowtimesParser class (it needs to inherit from DOMParserBase)
* write the parsing rules using the piculet schema
* add the class reference to the _OBJECTS dict at the botton, calling the key `showtime_parser` so that it can be called as `self.showtimesParser.showtime_parser` from the `get_showtimes`

I'm sure it's all quite confusing. After all the code is almost 20 year old, and a massive refactory would help.

Anyway, the best thing is to copy from similar classes; if more help is needed, just ask.

**Comment by kostyafarber** [comment]
@alberanid awesome. Thanks for the reply and direction. I already got started on it and pretty much have built the parser (albeit for [this](https://www.imdb.com/showtimes/location?ref_=sh_lc) page), however I can easily transfer it to the page you referenced. 

I was able to get my head around for code after a little while 😄 

All the implementation details you've just mentioned are super helpful. I'll take a look at those and work on it and submit a PR later.

Thanks!

**Comment by kostyafarber** [comment]
@alberanid added a PR at https://github.com/cinemagoer/cinemagoer/pull/406. Let me know if I need to make any changes. Many thanks!

**Comment by alberanid** [comment]
hi!

I've merged the PR of @kostyafarber (thanks!) after some changes.

Now the `get_showtimes()` method returns a list of objects with information about the cinema and the movies.

An example of the output:
```
[
  {
    'cinema': 'Multisala Astra',
    'address and contacts': 'an address | a phone',
    'movies': [
        {
            'movie': MovieInstance,
            'showtimes': '4:20 pm | 6:50 | 9:15'
        },
        {...}
  },
  ...
]
```
